### PR TITLE
rst hooks: ignore code blocks

### PR DESCRIPTION
**Pull request by hugovk**
Follow on from https://github.com/pre-commit/pygrep-hooks/pull/29 and https://github.com/pre-commit/pygrep-hooks/pull/30.

We found a problem with using `rst-backticks` to lint PEPs (https://github.com/python/peps/pull/1668), code blocks containing single backticks are valid. For example, this is fine but the linter complains:

```rst
    class CompiledGenSeries:

        # This class is what the `gen_series()` generator can
        # be transformed to by a compiler like Cython.
```

These are called "literal blocks" in rst: https://docutils.sourceforge.io/docs/user/rst/quickref.html#literal-blocks

> A paragraph containing only two colons
indicates that the following indented
or quoted text is a literal block.

Where quoting is given as lines beginning `>`.

For our purposes, I think it's good enough to ignore the `::` bit and skip lines beginning with a space or `>`.

This PR prepends the regex with `^(?![ >]).*`

That is:
* start of line
* Negative lookahead for space or `>`
* Any other char

The same problem can potentially occur with `rst-inline-touching-normal`, so I added it there too.

### CODE REVIEW

File: tests/hooks_test.py [hunk 1]
```diff
@@ -185,6 +188,8 @@ def test_python_rst_inline_touching_normal_positive(s):
         '``inline`` normal',
         'normal ``inline`` normal',
         'normal ``inline``',
+        '    ``indented``inline touching',
+        '> quoted ``inline``touching',
```
Comment (asottile) [inline tests/hooks_test.py:192]: these still look incorrect and I think should be flagged

File: .pre-commit-hooks.yaml [hunk 2]
```diff
@@ -50,7 +50,7 @@
 -   id: rst-backticks
     name: rst ``code`` is two backticks
     description: 'Detect common mistake of using single backticks when writing rst'
-    entry: '(^| )`[^`]+`([^_]|$)'
+    entry: '^(?![ >]).*(^| )`[^`]+`([^_]|$)'
```
Comment (asottile) [inline .pre-commit-hooks.yaml:53]: this false-positives for

```
some text
 where someone `did this`
```

I think the case for this is exceedingly rare -- I don't think we should change the regex to accommodate this edge case

File: tests/hooks_test.py (hunk 1 above)
Comment (hugovk) [reply tests/hooks_test.py:192]: We didn't actually see any issues with `rst-inline-touching-normal` in https://github.com/python/peps/pull/1668/files, shall I just ditch the changes to this hook?

Comment (asottile) [reply tests/hooks_test.py:192]: yes

File: .pre-commit-hooks.yaml (hunk 2 above)
Comment (hugovk) [reply .pre-commit-hooks.yaml:53]: All the misflagged code block issues in https://github.com/python/peps/pull/1668/files began with four-space indents.

Shall I switch it to only skip for `^` + 4 spaces?

Comment (asottile) [reply .pre-commit-hooks.yaml:53]: yeah that seems fine

Comment (hugovk) [reply .pre-commit-hooks.yaml]: Thanks, updated/rebased!

**Review by asottile — APPROVED** [review]
